pkg/container/images/image.go

```diff
@@ -26,14 +26,21 @@ type ImageManager interface {
 // NewImageManager creates an instance of ImageManager appropriate
 // for the current environment, or returns an error if it is not supported.
 func NewImageManager(ctx context.Context) ImageManager {
-	// ASSUMPTION: This only works for the Docker runtime.
-	// Otherwise, return a no-op implementation.
+	// First try the registry-based implementation (no Docker daemon required)
```

Comment:
We need to return the no-op when we are inside k8s.

Reply:
That was the assumption before now. I think now we should explicitly check to see if we are operating inside k8s. (There is a `container.IsKubernetesRuntime()` method for this)